objects/src/accounts/storage/mod.rs

```diff
@@ -1,450 +1,239 @@
-        for item in items {
-            if item.index == AccountStorage::SLOT_LAYOUT_COMMITMENT_INDEX {
-                return Err(AccountError::StorageSlotIsReserved(item.index));
-            }
-
-            if matches!(item.slot.slot_type, StorageSlotType::Map { .. }) {
-                // check that for every map index there is a map in maps
-                if !maps.contains_key(&item.index) {
-                    return Err(AccountError::StorageMapNotFound(item.index));
-                }
-                num_maps += 1;
-            }
-
-            layout[usize::from(item.index)] = item.slot.slot_type;
-            entries.push((item.index.into(), item.slot.value))
-        }
+    pub fn new(slots: Vec<StorageSlot>) -> Result<AccountStorage, AccountError> {
+        let len = slots.len();
 
-        // add layout commitment entry
-        entries.push((
-            AccountStorage::SLOT_LAYOUT_COMMITMENT_INDEX.into(),
-            *layout_commitment(&layout),
-        ));
-
-        // construct storage slots smt and populate the types vector.
-        let slots = SimpleSmt::<STORAGE_TREE_DEPTH>::with_leaves(entries)
-            .map_err(AccountError::DuplicateStorageItems)?;
-
-        // make sure the number of provide maps matches the number of map slots
-        if maps.len() != num_maps {
-            return Err(AccountError::StorageMapTooManyMaps {
-                expected: num_maps,
-                actual: maps.len(),
-            });
+        if len > Self::NUM_STORAGE_SLOTS {
+            return Err(AccountError::StorageTooManySlots(len as u64));
         }
 
-        Ok(Self { slots, layout, maps })
+        Ok(Self { slots })
     }
 
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
+    /// Returns a reference to the storage slots.
+    pub fn slots(&self) -> &Vec<StorageSlot> {
+        &self.slots
+    }
+
     /// Returns a commitment to this storage.
-    pub fn root(&self) -> Digest {
-        self.slots.root()
+    pub fn commitment(&self) -> Digest {
+        build_slots_commitment(&self.slots)
     }
 
-    /// Returns an item from the storage at the specified index.
+    /// Converts storage slots of this [AccountStorage] into a vector of field elements.
     ///
-    /// If the item is not present in the storage, [crate::EMPTY_WORD] is returned.
-    pub fn get_item(&self, index: u8) -> Digest {
-        let item_index = NodeIndex::new(Self::STORAGE_TREE_DEPTH, index.into())
-            .expect("index is u8 - index within range");
-        self.slots.get_node(item_index).expect("index is u8 - index within range")
+    /// This is done by first converting each procedure into exactly 8 elements as follows:
+    /// ```text
+    /// [STORAGE_SLOT_VALUE, storage_slot_type, 0, 0, 0]
+    /// ```
+    /// And then concatenating the resulting elements into a single vector.
+    pub fn as_elements(&self) -> Vec<Felt> {
```

Comment:
The main benefit of this would be to have the simple way to get storage for most accounts while handling edge cases for accounts that use storage maps separately. For example, when we execute/prove transactions now, we need to load full account storage. This is fine when storage is small, but if storage is many MB - this won't work.

What we could do instead is load only the account storage "header" and for vast majority of transactions that should be enough. For the ones which have storage maps, we'd have a separate way of retrieving values from the  key-value maps (e.g., via the `DataStore` trait).

Reply:
Should we update the `AccountStorage` to store this header? 

where this would be the updated version: 

```rust
pub struct AccountStorage {
        header: AccountStorageHeader,
        slots: Vec<StorageSlot>
}
```

Reply:
Understood, issue opened here: https://github.com/0xPolygonMiden/miden-base/issues/867